ttl/ttlworker/job_manager.go

```diff
@@ -491,104 +407,34 @@ func (m *JobManager) rescheduleJobs(se session.Session, now time.Time) {
 	}
 	if !variable.EnableTTLJob.Load() {
 		if len(m.runningJobs) > 0 {
-			ctx, cancel := context.WithTimeout(m.ctx, ttlInternalSQLTimeout)
-
 			for _, job := range m.runningJobs {
-				logutil.Logger(m.ctx).Info("cancel job because tidb_ttl_job_enable turned off", zap.String("jobID", job.id), zap.String("statistics", job.statistics.String()))
-				m.removeJob(job)
-				err := job.Cancel(ctx, se)
-				if err != nil {
-					logutil.Logger(m.ctx).Warn("fail to cancel job", zap.Error(err))
-				}
-				job.finish(se, se.Now())
+				logutil.Logger(m.ctx).Info("cancel job because tidb_ttl_job_enable turned off", zap.String("jobID", job.id))
+				// TODO: cancel job
```

Comment:
Is the TODO here means we should cancel the job for other cases? For example: table deleted, table ttl option removed...

Reply:
> For example: table deleted, table ttl option removed...

They are not big problems, as the scan task will fail right? If a single batched scan query is not too big, it will be OK. (It makes me thinking another problem: as the `limit` is limiting the result set, it actually cannot limit the scanning range of a query. If the ttl column has some correlation with the primary key, the query size of scan query can be unpredictable).

> Is the TODO here means we should cancel the job for other cases? 

We could cancel job for **this** case, as the worker will not return error or somehow stop the task. We could also add this condition in the scan task manager, so that it can stop the ttl tasks directly (and return an error), without distributed cancelling (which hasn't been implemented). Do you think it's a good idea?

Reply:
When `tidb_ttl_job_enable` is disabled, the task will also fail automatically

Reply:
Done. Added some logic to remove the tasks, and cancel the tasks in the task manager.